src/data/nat/fib.lean

```diff
@@ -0,0 +1,99 @@
+/-
+Copyright (c) 2019 Kevin Kappelmann. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kevin Kappelmann
+-/
+import ..stream.basic
+import .....tactic.norm_num
+import .....tactic.monotonicity
+/-!
+# The Fibonacci Sequence
+
+## Summary
+
+Definition of the Fibonacci sequence `F₀ = 0, F₁ = 1, Fₙ₊₂ = Fₙ + Fₙ₊₁`.
+
+## Main Definitions
+
+- `fib` returns the stream of Fibonacci numbers.
+
+## Main Statements
+
+- `fib.recurrence` shows that `fib` indeed satisfies the Fibonacci recurrence `Fₙ₊₂ = Fₙ + Fₙ₊₁.`.
+
+## Implementation Notes
+
+For efficiency purposes, the sequence is defined using `stream.iterate`.
+
+## Tags
+
+fib, fibonacci
+-/
+
+private def fib.aux_stream : stream (ℕ × ℕ) := stream.iterate (λ p, ⟨p.snd, p.fst + p.snd⟩) ⟨0, 1⟩
+
+/--
+Implementation of the fibonacci sequence satisfying
+`fib 0 = 0, fib 1 = 1, fib (n + 2) = fib n + fib (n + 1)`.
+
+*Note:* We use a stream iterator for better performance when compared to the naive recursive
+implementation.
+-/
+def fib (n : ℕ) : ℕ := (fib.aux_stream n).fst
+
+namespace fib
+
+@[simp]
+lemma zeroth : fib 0 = 0 := rfl
```

Comment:
Should be called `fib_zero`. Also, `@[simp]` should be on same line

Reply:
Fixed.